packages/headless/src/features/advanced-search-queries/advanced-search-query-selectors.test.ts

```diff
@@ -0,0 +1,141 @@
+        aq: 'aq-parameter-value',
+      };
+      const state = {advancedSearchQueries: exampleAdvancedSearchQueries};
+
+      const result = selectAdvancedSearchQueries(state);
+
+      expect(result).toEqual(expectedOutput);
+    });
+
+    test("should return only not empty properties (here: 'cq') from the state", () => {
+      const exampleAdvancedSearchQueries: AdvancedSearchQueriesState =
+        createAdvancedSearchQueriesState(
+          'cq' as AdvancedSearchQueryParametersSelectedProperty,
+          'cq-parameter-value'
+        );
+      const expectedOutput = {
+        cq: 'cq-parameter-value',
+      };
+      const state = {advancedSearchQueries: exampleAdvancedSearchQueries};
+
+      const result = selectAdvancedSearchQueries(state);
+
+      expect(result).toEqual(expectedOutput);
+    });
+
+    test("should return only not empty properties (here: 'dq') from the state", () => {
+      const exampleAdvancedSearchQueries: AdvancedSearchQueriesState =
+        createAdvancedSearchQueriesState(
+          'dq' as AdvancedSearchQueryParametersSelectedProperty,
+          'dq-parameter-value'
+        );
+      const expectedOutput = {
+        dq: 'dq-parameter-value',
+      };
+      const state = {advancedSearchQueries: exampleAdvancedSearchQueries};
+
+      const result = selectAdvancedSearchQueries(state);
+
+      expect(result).toEqual(expectedOutput);
+    });
+
+    test("should return only not empty properties (here: 'lq') from the state", () => {
+      const exampleAdvancedSearchQueries: AdvancedSearchQueriesState =
+        createAdvancedSearchQueriesState(
+          'lq' as AdvancedSearchQueryParametersSelectedProperty,
+          'lq-parameter-value'
+        );
+      const expectedOutput = {
+        lq: 'lq-parameter-value',
+      };
+      const state = {advancedSearchQueries: exampleAdvancedSearchQueries};
+
+      const result = selectAdvancedSearchQueries(state);
+
+      expect(result).toEqual(expectedOutput);
+    });
+  });
+});
+
+function newExampleAdvancedSearchQueriesStateFixture() {
+  return {
+    aq: 'aq-parameter-value',
+    cq: 'cq-parameter-value',
+    dq: 'dq-parameter-value',
+    lq: 'lq-parameter-value',
+    aqWasSet: true,
+    cqWasSet: false,
+    dqWasSet: false,
+    lqWasSet: true,
+    defaultFilters: {
+      aq: 'a-default-aq-parameter-value',
+      cq: 'a-default-cq-parameter-value',
+      dq: 'a-default-dq-parameter-value',
+      lq: 'a-default-lq-parameter-value',
+    },
+  };
+}
+
+function createAdvancedSearchQueriesState(
+  propertyUnderTest: AdvancedSearchQueryParametersSelectedProperty,
+  valueSet: string
+): AdvancedSearchQueriesState {
+  const state = {
+    aq: '',
+    cq: '',
+    dq: '',
+    lq: '',
+    aqWasSet: false,
+    cqWasSet: false,
+    dqWasSet: false,
+    lqWasSet: false,
+    defaultFilters: {
+      aq: '',
+      cq: '',
+      dq: '',
+      lq: '',
+    },
+  };
+
+  state[propertyUnderTest] = valueSet;
+
+  return state;
+}
```

Comment:
If you prefer you can use this pattern:

```suggestion
function (
  overrides: Partial<AdvancedSearchQueriesState>
): AdvancedSearchQueriesState {
    const defaultAdvancedSearchQueriesState = {
        aq: '',
        cq: '',
        dq: '',
        lq: '',
        aqWasSet: false,
        cqWasSet: false,
        dqWasSet: false,
        lqWasSet: false,
        defaultFilters: {
        aq: '',
        cq: '',
        dq: '',
        lq: '',
        },
    };
    return {
        ...defaultAdvancedSearchQueriesState,
        ...overrides
    };
}
```

Reply:
I like that! thanks!